main/src/main/scala/sbt/Defaults.scala

```diff
@@ -377,6 +397,13 @@ object Defaults extends BuildCommon {
         else Nil
       bases.map(b => new Source(b, include, exclude)) ++ baseSources
     },
+    watchProjectSources in ConfigGlobal := (watchProjectSources in ConfigGlobal).value ++ {
```

Comment:
Is it sufficient to `Global / watchProjectSources := Nil` to opt out of this?

Reply:
Should be. I am planning on overhauling the documentation before the 1.3.0 release to explain all of this.